src/pages/core/guides/onchain-ticketing-with-appdata.md

```diff
@@ -0,0 +1,512 @@
+       bump = manager.bump
+   )]
+   pub manager: Account<'info, Manager>,
+   #[account(
+       mut,
+       constraint = event.update_authority == manager.key(),
+   )]
+   pub event: Account<'info, BaseCollectionV1>,
+   #[account(mut)]
+   pub ticket: Signer<'info>,
+   pub system_program: Program<'info, System>,
+   #[account(address = MPL_CORE_ID)]
+   /// CHECK: This is checked by the address constraint
+   pub mpl_core_program: UncheckedAccount<'info>
+}
+```
+
+The main differences in the account struct are:
+- The event account is already initialized so we can deserialize it as a `BaseCollectionV1` asset where we can check that the `update_authority` is the manager PDA. 
+- The ticket account, set as mutable and a signer, will be transformed into a Core Collection Account during this instruction.
+
+Since we need to save extensive data in this function too, we pass these inputs via a structured format as done already in the `create_event` instruction.
+
+```rust
+#[derive(AnchorDeserialize, AnchorSerialize)]
+pub struct CreateTicketArgs {
+   pub name: String,
+   pub uri: String,
+   pub hall: String,
+   pub section: String,
+   pub row: String,
+   pub seat: String,
+   pub price: u64,
+   pub venue_authority: Pubkey,
+}
+```
+
+When we talk about the instruction, the main differences are:
+- Incorporates additional plugins like the `PermanentFreeze`, `PermanentBurn`, and `PermanentTransfer`in order to add a security layer in case something goes wrong. 
+- Use the new `AppData` external plugin to store binary data inside of it managed by the `venue_authority` that we pass in as input in the instruction. 
+- It has a sanity check at the start to see if the total number of ticket issued doesn’t go beyond capacity limit
+
+```rust
+pub fn create_ticket(ctx: Context<CreateTicket>, args: CreateTicketArgs) -> Result<()> {
+    // Check that the maximum number of tickets has not been reached yet
+    let (_, collection_attribute_list, _) = fetch_plugin::<BaseCollectionV1, Attributes>(
+            &ctx.accounts.event.to_account_info(), 
+            PluginType::Attributes
+        )?;
+
+    // Search for the Capacity attribute
+    let capacity_attribute = collection_attribute_list
+        .attribute_list
+        .iter()
+        .find(|attr| attr.key == "Capacity")
+        .ok_or(TicketError::MissingAttribute)?;
+
+    // Unwrap the Capacity attribute value
+    let capacity = capacity_attribute
+        .value
+        .parse::<u32>()
+        .map_err(|_| TicketError::NumericalOverflow)?;
+
+    require!(
+        ctx.accounts.event.num_minted < capacity, 
+        TicketError::MaximumTicketsReached
+    );
+
+    // Add an Attribute Plugin that will hold the ticket details
+    let mut ticket_plugin: Vec<PluginAuthorityPair> = vec![];
+    
+    let attribute_list: Vec<Attribute> = vec![
+    Attribute { 
+        key: "Ticket Number".to_string(), 
+        value: ctx.accounts.event.num_minted.checked_add(1).unwrap().to_string() 
```

Comment:
for checked_add(1) you could probably do `ok_or(TicketError::NumericalOverflow)?`